### PR TITLE
TypeScript | Composition API | Unit Tests

### PR DESCRIPTION
**Pull request by WawrzenskiPatryk**
⚠️ PR's calulated to be equal around **2000 lines** change, but it's **much smaller**. The amount of calculated lines is caused by including new packages to the project - check `package-lock.json`.

Overall this PR provides no new functionality in terms of client's perspective, only developers' one. It is mostly refactor of already existing code with usage of TypeScript and Composition API / script setup or unit test files. The refactor includes only a few files so I could take some first steps in mentioned fields without tweaking the whole project.

### CODE REVIEW

**Comment by WawrzenskiPatryk** [comment]
A bunch of changes incoming since adding a change in `tsconfig.json` file
```json
{
  "compilerOptions": {
    // ...
    "strict": true
  }
}
```
Which caused some extra scenario handling requirements 😄

File: src/store/auth.ts [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
+import { login, register, setAuthHeader, removeAuthHeader } from '@/services/api/auth';
+import { LoginCredentials, RegisterCredentials } from '@/types/auth';
 
 const TOKEN_STORAGE_KEY = 'auth-token';
 
 export const useAuthStore = defineStore('auth', {
   state() {
     return {
-      token: null,
+      token: null as string,
```
Comment (szymonlicau) [inline src/store/auth.ts]: Null isn't a string - you could set it by default as an empty string (it's a falsy value) or perhaps mark this as `string | null` 😄 
```suggestion
      token: '',
```

File: src/store/index.ts [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
+import { GenreData, MovieData, SeanceData, ScreeningData } from '@/types/data';
 import router from '@/router';
 
 export const useMainStore = defineStore('main', {
   state() {
     return {
-      error: null,
-      errorDisplayTime: 4,
+      error: null as Error,
+      errorDisplayTime: 4 as number,
```
Comment (szymonlicau) [inline src/store/index.ts]: Basic types will be inferred without issues 😎 
```suggestion
      errorDisplayTime: 4,
```

File: src/store/index.ts [hunk 3]
```diff
@@ -1,25 +1,26 @@
...
       areSeancesLoading: false,
       areScreeningsLoading: false,
-      allMovies: [],
-      allGenres: [],
-      storedSeances: [],
-      storedScreenings: [],
+      allMovies: [] as Array<MovieData>,
+      allGenres: [] as Array<GenreData>,
+      storedSeances: [] as Array<SeanceData>,
+      storedScreenings: [] as Array<ScreeningData>,
```
Comment (szymonlicau) [inline src/store/index.ts]: You might like to know that you can also use the shorthand for this if you like to 😄 
```suggestion
      allMovies: [] as MovieData[],
      allGenres: [] as GenreData[],
      storedSeances: [] as SeanceData[],
      storedScreenings: [] as ScreeningData[],
```

File: tests/unit/services/dates.test.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      expect(getWeekdayName(5)).toBe('Friday');
+      expect(getWeekdayName(6)).toBe('Saturday');
+    });
+
+    it('throws an error for id outside the range of 0 to 6', () => {
+      expect(() => getWeekdayName(-3)).toThrowError();
+      expect(() => getWeekdayName(-1)).toThrowError();
+      expect(() => getWeekdayName(7)).toThrowError();
+      expect(() => getWeekdayName(8)).toThrowError();
+      expect(() => getWeekdayName(12)).toThrowError();
```
Comment (szymonlicau) [inline tests/unit/services/dates.test.js]: No need to provide more than the first edge-case examples 😄 
```suggestion
      expect(() => getWeekdayName(-1)).toThrowError();
      expect(() => getWeekdayName(7)).toThrowError();
```

File: src/store/index.ts [hunk 5]
```diff
@@ -1,25 +1,26 @@
...
 import { getAllMoviesData, getMovieGenresData, getSeancesData } from '@/services/api/data.js';
+import { GenreData, MovieData, SeanceData, ScreeningData } from '@/types/data';
 import router from '@/router';
 
 export const useMainStore = defineStore('main', {
   state() {
     return {
-      error: null,
-      errorDisplayTime: 4,
+      error: null as Error,
```
Comment (szymonlicau) [inline src/store/index.ts]:
```suggestion
      error: null as Error | null,
```

**Review by szymonlicau — CHANGES_REQUESTED** [review]
Nice first steps to automated testing, looking good!
I have a few small suggestions in regards to TypeScript and the tests but overall really nice work! ❤️

File: src/store/auth.ts (hunk 1 above)
Comment (WawrzenskiPatryk) [reply src/store/auth.ts]: About to be `''` 😄

File: src/store/index.ts (hunk 2 above)
Comment (WawrzenskiPatryk) [reply src/store/index.ts]: Right, i've added it because TS had problems with `this` in store, but it's solved now as u know 😄

File: src/store/index.ts (hunk 5 above)
Comment (WawrzenskiPatryk) [reply src/store/index.ts]: Yeah, after adding `"strict": true` to the config file, lDE complains about it as well, so it really helps a lot!

**Review by szymonlicau — APPROVED** [review]
